### PR TITLE
createCollection() SDK method

### PR DESCRIPTION
**Pull request by cedric25**

### CODE REVIEW

File: packages/sdk/tests/e2e/sharing/createCollection.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  describe('When calling createCollection()', () => {
+    it(
+      'should answer with a collection address',
+      async () => {
+        const { collectionId } = await dataProtector.createCollection();
+        expect(collectionId).toBeDefined();
+      },
+      2 * MAX_EXPECTED_BLOCKTIME
```
Comment (cedric25) [inline packages/sdk/tests/e2e/sharing/createCollection.test.ts:22]: Check if good timeout value here.

File: packages/sdk/src/dataProtector/types.ts [hunk 2]
```diff
@@ -348,6 +348,9 @@ export type TransferResponse = {
   txHash: string;
 };
 
+// Sharing methods
+export type { CreateCollectionResponse } from "./sharing/createCollection.js"
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/types.ts:352]: Trying something here to slow down this file length: the method file will own its types, just re-exporting them here.

File: packages/sdk/src/dataProtector/sharing/createCollection.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { ethers, type Contract } from 'ethers';
+import { ABI as sharingABI } from '../../contracts/sharingAbi.js';
+import { WorkflowError } from '../../utils/errors.js';
+import { throwIfMissing } from '../../utils/validators.js';
+import { AddressOrENS, IExecConsumer } from '../types.js';
+
+export type CreateCollectionResponse = {
+  collectionId: AddressOrENS;
+};
```
Comment (cedric25) [inline packages/sdk/src/dataProtector/sharing/createCollection.ts:9]: Trying something here to slow down `types.ts` file length: the method file will own its types as we can see here.

File: packages/sdk/src/dataProtector/sharing/createCollection.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { ethers, type Contract } from 'ethers';
+import { ABI as sharingABI } from '../../contracts/sharingAbi.js';
+import { WorkflowError } from '../../utils/errors.js';
+import { throwIfMissing } from '../../utils/validators.js';
+import { AddressOrENS, IExecConsumer } from '../types.js';
+
+export type CreateCollectionResponse = {
+  collectionId: AddressOrENS;
```
Comment (PierreJeanjacquot) [inline packages/sdk/src/dataProtector/sharing/createCollection.ts:8]: collectionId will be a number

Comment (cedric25) [reply packages/sdk/src/dataProtector/sharing/createCollection.ts]: Absolutely ✅

File: packages/sdk/src/dataProtector/sharing/createCollection.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      );
+    });
+
+  const mintedTokenId = transactionReceipt.logs.find(
+    ({ eventName }) => eventName === 'Transfer'
+  )?.args[2];
+
+  return {
+    collectionId: mintedTokenId,
+  };
```
Comment (PierreJeanjacquot) [inline packages/sdk/src/dataProtector/sharing/createCollection.ts:41]: As tokenid is an uint256 in the ABI `mintedTokenId` is going to be a bigint. Assuming we are using a sequence for collectionIds, we can safely cast it into a number that will fit into safe integers.

```suggestion
  const mintedTokenId = transactionReceipt.logs.find(
    ({ eventName }) => eventName === 'Transfer'
  )?.args[2];

  return {
    collectionId: Number(mintedTokenId),
  };
```

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
good job, see the comments for improvement suggestions

File: packages/sdk/src/dataProtector/sharing/createCollection.ts (hunk 5 above)
Comment (cedric25) [reply packages/sdk/src/dataProtector/sharing/createCollection.ts:41]: Would it make sense to add `as bigint` line 37 to better understand the Number casting line 40?

Comment (PierreJeanjacquot) [reply packages/sdk/src/dataProtector/sharing/createCollection.ts:41]: yes! It will make it clearer

**Pull request merged**